File: src/main/java/at/hrechny/predictionsbot/model/Competition.java
package at.hrechny.predictionsbot.model;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Null;
import java.util.UUID;
import lombok.Data;

@Data
public class Competition {

  @Null
  private UUID id;

  @NotNull
  private String name;

  @NotNull
  private Long apiFootballId;

  @NotNull
  private boolean active;

}

File: src/main/kotlin/at/hrechny/predictionsbot/model/Competition.kt
package at.hrechny.predictionsbot.model

import io.micronaut.core.annotation.Introspected
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Null
import java.util.UUID

@Introspected
class Competition {
    @field:Null
    var id: UUID? = null

    @field:NotNull
    var name: String? = null

    @field:NotNull
    var apiFootballId: Long? = null

    @field:NotNull
    var active: Boolean = false

    fun isActive(): Boolean = active
}
